### DISCUSSION TITLE
Swift Playgrounds Support!

### DISCUSSION THREAD

**Discussion by orchetect** [orchetect/swift-midi#122] [Show and tell]
MIDIKit works in Swift Playgrounds on both iPad and macOS!

The most impressive part is that you don't even need to run the app. The live preview is fully interactive and can send and receive MIDI.

![midikit-playgrounds](https://user-images.githubusercontent.com/11605557/175794808-46d2f03e-1717-4d73-aabd-9bf6a8f15219.png)

**Comment by JamesGreenaway** [comment]
@orchetect I can't get other apps to recognise my virtual MIDI output when running this app using the example code given at iOS SwiftUI > Virtual Output. It works when I build using Xcode, however, running in Swift Playgrounds on iPad, nothing happens. Any suggestions?

**Reply by JamesGreenaway** [reply]
The USBiOStoMac example also works fine for Swift Playgrounds

**Reply by orchetect** [reply]
A virtual output requires the background audio entitlement for iOS. I’m not sure how that would work for Swift Playgrounds.

**Reply by JamesGreenaway** [reply]
I see, ok thanks for responding. 🙏
